Bug: 2.7.1 break faker locale generator

### Description

After upgrading from 2.7.0 to 2.7.1 my tests fail due to a change in how faker is used.
Error:
```
/factories.py:93: in vat_identifier
    return cls.__faker__.vat_id()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <faker.proxy.Faker object at 0x103626ad0>, attr = 'vat_id'

    def __getattr__(self, attr: str) -> Any:
        """
        Handles cache access and proxying behavior
    
        :param attr: attribute name
        :return: the appropriate attribute
        """
        if len(self._factories) == 1:
>           return getattr(self._factories[0], attr)
E           AttributeError: 'Generator' object has no attribute 'vat_id'
```

### URL to code causing the issue

_No response_

### MCVE

```python
# Works in 2.7.0 not in 2.7.1:
class SellerFactory(ModelFactory[Seller]):
    __model__ = Seller
    __set_as_default_factory_for_type__ = True
    __faker__ = Faker(locale="de_DE")
    __random_seed__ = 10

    @classmethod
    def vat_identifier(cls) -> str:
        return cls.__faker__.vat_id()
```


### Steps to reproduce

```bash
1. Go to '...'
2. Click on '....'
3. Scroll down to '....'
4. See error
```


### Screenshots

"In the format of: `![SCREENSHOT_DESCRIPTION](SCREENSHOT_LINK.png)`"


### Logs

_No response_

### Release Version

2.7.1

### Platform

- [ ] Linux
- [X] Mac
- [ ] Windows
- [ ] Other (Please specify in the description above)

<!-- POLAR PLEDGE BADGE START -->
---

## Funding
* If you would like to see an issue prioritized, make a pledge towards it!
* We receive the pledge once the issue is completed & verified

<a href="https://polar.sh/litestar-org/polyfactory/issues/330">
<picture>
  <source media="(prefers-color-scheme: dark)" srcset="https://polar.sh/api/github/litestar-org/polyfactory/issues/330/pledge.svg?darkmode=1">
  <img alt="Fund with Polar" src="https://polar.sh/api/github/litestar-org/polyfactory/issues/330/pledge.svg">
</picture>
</a>
<!-- POLAR PLEDGE BADGE END -->

--- a/docs/examples/configuration/test_example_3.py
+++ b/docs/examples/configuration/test_example_3.py
@@ -26,4 +26,4 @@ class PersonFactory(DataclassFactory[Person]):
 
 def test_setting_faker() -> None:
     # the outcome of faker deterministic because we seeded random, and it uses a spanish locale.
-    assert PersonFactory.build().name == "Jessica Lane"
+    assert PersonFactory.build().name == "Alejandra Romeu-Tolosa"

--- a/polyfactory/factories/base.py
+++ b/polyfactory/factories/base.py
@@ -402,10 +402,7 @@ class BaseFactory(ABC, Generic[T]):
 
         """
         cls.__random__ = Random(seed)
-
-        faker = Faker()
-        faker.seed_instance(seed)
-        cls.__faker__ = faker
+        cls.__faker__.seed_instance(seed)
 
     @classmethod
     def is_ignored_type(cls, value: Any) -> bool:

--- a/tests/test_random_configuration.py
+++ b/tests/test_random_configuration.py
@@ -3,6 +3,7 @@ from random import Random
 from typing import cast
 
 import pytest
+from faker import Faker
 
 from polyfactory.factories.dataclass_factory import DataclassFactory
 
@@ -90,3 +91,19 @@ def test_setting_random_seed_on_multiple_factories() -> None:
 
     assert FooFactory.build().foo == RANDINT_MAP[foo_seed]
     assert BarFactory.build().bar == RANDINT_MAP[bar_seed]
+
+
+def test_no_override_of_faker() -> None:
+    faker = Faker()
+
+    @dataclass
+    class Foo:
+        foo: int
+
+    class FooFactory(DataclassFactory[Foo]):
+        __model__ = Foo
+
+        __faker__ = faker
+        __random_seed__ = 10
+
+    assert FooFactory.__faker__ is faker

--- a/tests/test_random_configuration.py
+++ b/tests/test_random_configuration.py
@@ -1,13 +1,22 @@
 from dataclasses import dataclass
 from random import Random
-from typing import cast
+from typing import List, Union, cast
 
 import pytest
 from faker import Faker
+from faker.config import AVAILABLE_LOCALES
 
 from polyfactory.factories.dataclass_factory import DataclassFactory
 
+FakerLocaleType = Union[str, List[str]]
+
 RANDINT_MAP = {i: Random(i).randint(0, 10) for i in range(3)}
+FAKER_LOCALES: List[FakerLocaleType] = [
+    AVAILABLE_LOCALES[0],
+    AVAILABLE_LOCALES[1],
+    AVAILABLE_LOCALES[2],
+    AVAILABLE_LOCALES[:3],
+]
 
 
 @pytest.mark.parametrize("seed", RANDINT_MAP.keys())
@@ -107,3 +116,20 @@ def test_no_override_of_faker() -> None:
         __random_seed__ = 10
 
     assert FooFactory.__faker__ is faker
+
+
+@pytest.mark.parametrize("locale", FAKER_LOCALES)
+def test_faker_locale_preserved(locale: FakerLocaleType) -> None:
+    @dataclass
+    class Foo:
+        foo: int
+
+    class FooFactory(DataclassFactory[Foo]):
+        __model__ = Foo
+
+        __faker__ = Faker(locale=locale)
+        __random_seed__ = 10
+
+    expected_locales = [locale] if isinstance(locale, str) else locale
+
+    assert FooFactory.__faker__.locales == expected_locales

